pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1113,6 +1099,21 @@ func (og *operationGenerator) GenerateUnmapDeviceFunc(
 			}
 		}
 
+		// Execute tear down device
+		unmapErr := blockVolumeUnmapper.TearDownDevice(globalMapPath, deviceToDetach.DevicePath)
+		if unmapErr != nil {
+			// On failure, return error. Caller will log and retry.
+			return deviceToDetach.GenerateError("UnmapDevice.TearDownDevice failed", unmapErr)
+		}
+
+		// Plugin finished TearDownDevice(). Now globalMapPath dir and plugin's stored data
+		// on the dir are unnecessary, clean up it.
+		removeMapPathErr := og.blkUtil.RemoveMapPath(globalMapPath)
+		if removeMapPathErr != nil {
+			// On failure, return error. Caller will log and retry.
+			return deviceToDetach.GenerateError("UnmapDevice failed", removeMapPathErr)
```

Comment:
keep original msg `UnmapDevice.RemoveLoopDevice failed`

Reply:
@rootfs 

I changed from "UnmapDevice" to "UnmapDevice.RemoveMapPath" as suggested.
Also, I rebased the commit. PTAL